pkg/client/query/sessionquerier.go

```diff
@@ -86,10 +86,6 @@ func (sessq *sessionQuerier) GetSession(
 		return session, nil
 	}
 
-	// Use mutex to prevent multiple concurrent cache updates
-	sessq.sessionsMutex.Lock()
```

Comment:
This mechanism is necessary to prevent the **thundering herd** problem, which can degrade performance even more than the lock overhead itself by triggering multiple redundant queries.

In a highly concurrent scenario, we can easily encounter the following sequence:

```
p1: CheckCache->QuerySession------------------------->SessionFetched->PopulateCache
p2: -------------------------CheckCache->QuerySession
```

As a result, the full node is queried twice unnecessarily.
This occurs because, during the time it takes for `p1` to complete its query and populate the cache, `p2` still sees the cache as empty and issues its own query as well.